doc/specs/stdlib_stringlist_type.md

```diff
@@ -0,0 +1,379 @@
+  type(stringlist_index_type) :: index
+
+  index = fidx(1)
+
+  !> inserting 2 elements to the stringlist
+  call stringlist%insert_at( fidx(1), "Element No. one" )
+  call stringlist%insert_at( index, "Element No. two" )
+    ! stringlist <-- {"Element No. two", "Element No. one"}
+
+  index = bidx(3)
+
+  !> inserting 2 elements to the stringlist
+  call stringlist%insert_at( bidx(1), "Element No. three" )
+  call stringlist%insert_at( index, "Element No. four" )
+    ! stringlist <-- {"Element No. two", "Element No. four", 
+    "Element No. one", "Element No. three"}
+
+end program demo_fidx_bidx
+```
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Constructor for stringlist
+
+#### Description
+
+No arguments given: Initializes an empty stringlist(a list containing no elements in it).
+
+With argument: Initializes a stringlist equivalent to the input array `array` i.e. a stringlist containing all elements of the input array `array` in the same order.
+
+#### Syntax
+
+- No arguments given: `res = [[stdlib_stringlist_type(module):stringlist_type(interface)]] ()`
+
+- With argument: `res = [[stdlib_stringlist_type(module):stringlist_type(interface)]] (array)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+1). No arguments.
+
+2). With argument:
+- `array`: array of Character scalar or array of [[stdlib_string_type(module):string_type(type)]]. 
+This argument is intent(in).
+
+#### Result value
+
+The result is an instance of type `stringlist_type`.
+
+#### Example
+
+```fortran
+program demo_constructor
+  use stdlib_stringlist_type, only: stringlist_type
+  use stdlib_string_type, only: string_type
+  implicit none
+
+  type(stringlist_type) :: stringlist
+
+  stringlist = stringlist_type()
+    ! stringlist <-- { } (empty stringlist)
+
+  stringlist = stringlist_type(["#1", "#2", "#3"])
+    ! stringlist <-- {"#1", "#2", "#3"}
+
+  stringlist = stringlist_type([string_type("#1"), string_type("#2")])
+    ! stringlist <-- {"#1", "#2"}
+
+end program demo_constructor
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### insert_at
+
+#### Description
+
+Inserts the string `string` _AT_ the index `idx`, so that the newly added element is present at index `idx` after insertion. Inserting an element _AT_ index beyond `length + 1` inserts the element _AT_ `list_tail`, and likewise inserting _AT_ a non-positive index inserts the element _AT_ `list_head`.
+
+#### Syntax
+
+`call [[stdlib_stringlist_type(module):stringlist_type(type)]]%[[stdlib_stringlist_type(module):insert_at(generic)]] (idx, string)`
+
+#### Status
+
+Experimental.
+
+#### Class
+
+Pure subroutine.
+
+#### Argument
+
+- `idx`: [[stdlib_stringlist_type(module):stringlist_index_type(type)]].
+  This argument is intent(in).
+
+- `string`: Character scalar or [[stdlib_string_type(module):string_type(type)]].
+  This argument is intent(in).
+
+#### Result value
+
+No result.
```

Comment:
I don't agree with `No result`. Something happens. What about:

```suggestion
The result is the instance of type `strdlinglist_type` with an additional string.
```

Reply:
Here `No result` means that there is no output.
The subroutine here does not return anything as the output but it does what it is mentioned in the description of the functionality.

Reply:
I don't agree with you: there is an output: an "updated" instance of type "stringlist_type" with an additional string.

The instance is not the same before and after the call of this subroutine. This output is also represented by the `intent(inout)`.

Reply:
If we mention this can it likely cause a reader to misunderstand this subroutine as a function instead which is returning a new `stringlist`, and a user might end using a syntax like this:
`list = list%insert_at( fidx(10), "10th element")`